Dockerfile

```diff
@@ -23,170 +23,27 @@ RUN set -ex && \
         gperf \
         unzip
```

Comment:
looks like `patch` is missing (either from `patch` package or `build-essential`) 

I think we could reproduce go with the same set of packages as https://github.com/monero-project/monero/blob/a39b1d56c8835798e2f5e3cc43c33b5f2d8e13da/.github/workflows/depends.yml#L77 - wdyt?

Reply:
(although `ccache` would not really be useful unless we use https://github.com/moby/buildkit/blob/e07f3883e343ab311ca96be8ca9be3619ed27db7/frontend/dockerfile/docs/syntax.md#run---mounttypecache)

Reply:
done